### ISSUE TITLE
Markdown support in Comments

### ISSUE THREAD

**Issue by sunjam** [nextcloud/server#17375]
Did not see this request previously submitted, so requesting Markdown support in official Comments app.  Markdown support is available in a lot of different apps, but notably absent from basic Comments.  Thanks for considering!
![Screen Shot 2019-10-02 at 10 23 13 AM](https://user-images.githubusercontent.com/1787238/66066796-14e73180-e4ff-11e9-8d5f-4882698f12b7.png)

edit: [Discussion about Comments here](https://help.nextcloud.com/t/lets-talk-about-comments/61238)

**Comment by juliushaertl** [comment]
Makes sense imo, although we should probably just have a reduced set of supported syntax. cc @nextcloud/designers

**Comment by trev-dev** [comment]
We use the comments to keep a record of conversation for collaboration, and the newlines are even stripped out. It makes it hard to say more than one or two sentences at a time. Would absolutely love markdown in comments.

**Comment by stefan-niedermann** [comment]
> reduced set of supported syntax

Which syntax would you remove and why?

**Comment by fthommen** [comment]
> reduced set of supported syntax

I can't see a benefit in a reduced set of supported syntax. There are already enough MD variants in the wild.  If MD should be supported, then I suggest to stick to an already existing variant or to https://daringfireball.net/projects/markdown/syntax.

**Comment by stefan-niedermann** [comment]
> I suggest to stick to an already existing variant or to https://daringfireball.net/projects/markdown/syntax.

If markdown should be supported then it makes **lots** of sense to use the already existing and established markdown parser from `Nextcloud Text` - Your proposal is nice but it would just introduce another set of supported syntax and is therefore not really an enhancement and introduces other issues compared to `Nextcloud Text`.

**Comment by fthommen** [comment]
well, as I said: "I suggest to stick to an already existing variant or ...". If NC has its own parser, then this _is_ an already existing variant :-)

**Comment by Dunrar** [comment]
Is this being worked on? The comments in the Deck app are not very usefull right now, compared to Trello.

**Comment by reos-rcrozier** [comment]
Increasing the allowed length of comments would also be useful.

**Comment by sunjam** [comment]
You'll want to file a different issue for increasing allowed length.

**Comment by susnux** [comment]
> Which syntax would you remove and why?

I would suggest to limit it to CommonMark and maybe even without heading support. As headings make no sense in comments. Especially every heading `h1` and `h2` will be wrong.

**Comment by gwnmikedus** [comment]
This would be lovely! It would be nice to have an in-card discussion available to for example provide screenshots/images. 

Scenario: Card is a bug 🐞 report, having the issue and screenshot in description. Another Deck member wants to include another screenshot to the comments and ask if that is related.

Loads of more scenarios the ability to have more than just flat text (blank lines are not possible either) available.


A workaround now is to use Talk with quick picking the card someone wants to mention on, but this is hard to find back later and keep an containerised overview of a specific "issue".

**Comment by sunjam** [comment]
Probably want to open a new issue or search for one involving attachments and screenshots, or use Deck repo issues to discuss that app. :heart:

**Comment by jancborchardt** [comment]
@juliushaertl possibly something for Office planning as a nice enhancement? But mostly if it also is used then in Deck comments like you said.

**Comment by juliushaertl** [comment]
For deck it was a straight forward implementation (https://github.com/nextcloud/deck/pull/6416), for files comments I'm not sure if they have other special logic, but maybe something for the files planning then

**Comment by jancborchardt** [comment]
> For deck it was a straight forward implementation ([nextcloud/deck#6416](https://github.com/nextcloud/deck/pull/6416)), for files comments I'm not sure if they have other special logic, but maybe something for the files planning then

FYI since the remaining part is Files comments, maybe it is as simple as for Deck? @susnux @artonge
